dbms/src/Processors/Formats/Impl/RegexpRowInputFormat.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <re2/re2.h>
+#include <string>
+#include <vector>
+#include <Core/Block.h>
+#include <Processors/Formats/IRowInputFormat.h>
+#include <Formats/FormatSettings.h>
+#include <Formats/FormatFactory.h>
+#include <IO/PeekableReadBuffer.h>
+
+namespace DB
+{
+
+class ReadBuffer;
+
+
+class RegexpRowInputFormat : public IRowInputFormat
```

Comment:
Need to add a comment with brief description of this format along with its options.